### PR TITLE
Update miner unit tests broken by #1290 and fix integration tests

### PR DESCRIPTION
**Pull request by Neurosploit**
After #1290 several unit tests failed related to the block assemblers. This PR fixes them.

### CODE REVIEW

**Comment by Neurosploit** [comment]
This also fixes the integration tests failing because of my posminting tests.

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs [hunk 1]
```diff
@@ -219,7 +219,7 @@ public void AddToPending(BlockPair blockPair)
         /// </summary>
         internal void ShutDown()
         {
-            this.asyncLoop.Dispose();
+            this.asyncLoop?.Dispose(); // null check in case loop was not started.
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs]: Was there a situation where this threw an exception @Neurosploit ? Just asking out of curiosity.

Also, the comment isn't needed ;)

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs [hunk 2]
```diff
@@ -250,7 +250,7 @@ public class ConsensusLoop : IConsensusLoop
         public void Stop()
         {
             this.Puller.Dispose();
-            this.asyncLoop.Dispose();
+            this.asyncLoop?.Dispose(); // null check in case loop was not started.
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs]: The comment isn't needed ;)

**Comment by bokobza** [comment]
Funnily enough, I managed to fix the tests by setting the timestamps in PowMiningTest.cs in my PR (https://github.com/stratisproject/StratisBitcoinFullNode/pull/1330).
I'm not against setting them up in multiple test classes though :-)

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs (hunk 1 above)
Comment (Neurosploit) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs]: @fassadlr Whilst I was working on the checkpoints PR I made some changes and an integration test failed which led me to find this scenario.

**Pull request merged**